### PR TITLE
[GR-57624] Fix Locale NPE in JDK 24+10 and better

### PR DESCRIPTION
**Pull request by jerboaa**
The JDK's StaticProperty class uses a default of the empty string for the java.locale.useOldISOCodes static property. Ensure to also use the empty string as the default in substratevm avoiding the NullPointerException.

Closes: #9526

Testing:
 - Passes the provided reproducer in #9526. Throws NPE before.
 - Quarkus tests from https://github.com/graalvm/mandrel/issues/778 which triggered this investigation.

### CODE REVIEW

**Comment by jerboaa** [comment]
@fniephaus Is this something the Oracle GraalVM team is aware of? Thanks!

**Comment by fniephaus** [comment]
Hi @jerboaa,
I don't think we are aware of this, but let me change that ;) Thanks for the PR and the ping!

**Comment by jerboaa** [comment]
@fniephaus Hi! Any update on this?

**Comment by fniephaus** [comment]
Thanks for the reminder, @jerboaa. I'm working on integrating this.

**Review by fniephaus — APPROVED** [review]

**Comment by jerboaa** [comment]
Thank you!

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java
Comment (fniephaus) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java:134]: On second thought, I think we should instead fix the `jdk.internal.util.StaticProperty.javaLocaleUseOldISOCodes()` substitution to return `SystemPropertiesSupport.singleton().savedProperties.getOrDefault("java.locale.useOldISOCodes", "")`, see https://github.com/oracle/graal/blob/9df286d2ac54eecbc66170c08e76ce23bcda9d34/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_jdk_internal_util_StaticProperty.java#L142

WDYT?

Comment (jerboaa) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java]: That works as well. Updated.

Comment (fniephaus) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java]: Turns out that @jovanstevanovic stumbled upon the same problem recently and is going to fix this with his next PR.

Comment (jerboaa) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java]: Fair enough. Please close #9526 once that PR merges.

Comment (fniephaus) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java]: I'm going to merge this to unblock Quarkus :)

Comment (jerboaa) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java]: That would be fantastic! :)

**Pull request merged**
